### ISSUE TITLE
Audio format being saved as video/webm

### ISSUE THREAD

**Issue by cheresanford** [OvidijusParsiunas/deep-chat#265]
Hello @OvidijusParsiunas ! First off thanks for the great package! I'm running on a little issue though.
I've implemented the deep-chat component on my Vue 3 app, as follows: 

**<deep-chat
      ref="deepChat"
      style="width: 100%; height: 100%"
      :audio=true
      :connect="chatConnection"
      :avatars="avatars"
      :textInput="textInputStyle"
      :submitButtonStyles="submitButtonStyles"
      :messageStyles="messageStyles"
      :requestInterceptor="requestInterceptor"
      :microphone="microphoneStyles"
      :history="chatHistory"
      @message="formatSpecificMessage"
    >
    </deep-chat>**
    
    
![imagem_2024-09-10_214300691](https://github.com/user-attachments/assets/fced36d4-57a6-436a-8340-c9ed2dd874f9)

On my frontend request it seems like the audio format is fine, but when I save it on my back-end (running locally) even though the extension shows as .mp3, if I upload the downloaded file into a file type checking website, it shows video/webm:
![imagem_2024-09-10_214528886](https://github.com/user-attachments/assets/032ab5f1-33da-498d-a4b7-5426eebc60da)
Which is also why my request to transcribe the audio file to ChatGPT isn't working with the downloaded file.

On my server side, utilizing Golang, nothing seems odd and I'm testing it pretty straightforward:

func GenerateChatCompletion(c *gin.Context) {
	isAudioMessage := false
	audioTranscription := ""
	// try to get formData files from the request
	files, header, _ := c.Request.FormFile("files")
	// if has files, save it and return the path
	if files != nil {
		isAudioMessage = true
		fileName := header.Filename

		// print header content-type
		print("Content-Type: ", header.Header.Get("Content-Type"), "\n")

		// File path
		var filePath string
		if os.Getenv("AWS_LAMBDA_RUNTIME_API") == "" {
			// Save the file to ./files directory
			filePath = filepath.Join("./files", fileName)
			destFile, err := os.Create(filePath)
			if err != nil {
				c.JSON(500, gin.H{
					"error":   true,
					"data":    nil,
					"message": "Failed to create file: " + err.Error(),
					"code":    500,
				})
				return
			}
			defer destFile.Close()
			}
			

Program ends after that with the file saved, but with video/webm MIME type though.

**Comment by delsoul6104** [comment]
I encountered the same issue. Even after setting the output format to MP3, when uploading to the backend and performing a format check, the file is still in WEBM format. The file name appears as MP3, but the actual format remains WEBM. Currently, I can only handle WEBM as the audio format for subsequent use.

**Comment by cheresanford** [comment]
> I encountered the same issue. Even after setting the output format to MP3, when uploading to the backend and performing a format check, the file is still in WEBM format. The file name appears as MP3, but the actual format remains WEBM. Currently, I can only handle WEBM as the audio format for subsequent use.

@delsoul6104 Hey, I have found the cause after a long while browsing the web. Seems the issue is not with DeepChat itself, but with the voice recording stuff they're using, which I believe to be [MediaStream Recording API](https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_Recording_API/Using_the_MediaStream_Recording_API). Seems like it's a going-on bug that's been around for a while, seen some Stackoverflow topics on this a few weeks ago.

So the solution for now is to use something like ffmpeg to manually convert the files if you really need MP3 format.

**Comment by OvidijusParsiunas** [comment]
Hi @cheresanford. Apologies for not getting back to you sooner as I missed quite a few issues while I was moving to a different country. As you mentioned, Deep Chat utilises the standard [MediaStream Recording API](https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_Recording_API/Using_the_MediaStream_Recording_API) that works across most browsers, the downside is that it can potentially come with a few bugs as per your experience. Unfortunately there is not much that can be done about this as it is out of Deep Chat's control and we'll have to wait for the actual browser developers who implement the standards to take care of the problem itself.
Apologies once again and fingers crossed it gets fixed soon!

**Comment by delsoul6104** [comment]
> > 我遇到了同樣的問題。即使將輸出格式設置為 MP3 後，在上傳到後端並執行格式檢查時，檔仍為 WEBM 格式。檔名顯示為 MP3，但實際格式仍為 WEBM。目前我只能處理 WEBM 作為音訊格式，供後續使用。
> 
> 嘿，我在流覽了很長時間后找到了原因。似乎問題不在於 DeepChat 本身，而在於他們使用的錄音功能，我認為是 [MediaStream Recording API](https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_Recording_API/Using_the_MediaStream_Recording_API)。似乎這是一個已經存在了一段時間的錯誤，幾周前看到了一些關於此的 Stackoverflow 主題。
> 
> 所以現在的解決方案是使用ffmpeg之類的東西來手動轉換檔，如果你真的需要MP3格式。




> > I encountered the same issue. Even after setting the output format to MP3, when uploading to the backend and performing a format check, the file is still in WEBM format. The file name appears as MP3, but the actual format remains WEBM. Currently, I can only handle WEBM as the audio format for subsequent use.
> 
> @delsoul6104 Hey, I have found the cause after a long while browsing the web. Seems the issue is not with DeepChat itself, but with the voice recording stuff they're using, which I believe to be [MediaStream Recording API](https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_Recording_API/Using_the_MediaStream_Recording_API). Seems like it's a going-on bug that's been around for a while, seen some Stackoverflow topics on this a few weeks ago.
> 
> So the solution for now is to use something like ffmpeg to manually convert the files if you really need MP3 format.

That's correct. Currently, if I need to use the MP3 format, I perform the conversion on the backend.
